posts/post.model.js

```diff
@@ -22,7 +22,10 @@ const postSchema = new mongoose.Schema(
     toJSON: {
       transform: function (doc, ret) {
         ret.author = ret.isAnonymous ? undefined : ret.author;
-        if (ret.Voted === false) {
+        if(ret.ownedByCurrentUser){
+          ret.Voted = undefined
+        }
+        if (ret.Voted === false && ret.ownedByCurrentUser === false) {
```

Comment:
this works fine, however, having a dedicated if condition for the that logic will simplify the code flow and make it more readable, usually we check edge cases in separate if(s)